hyperhtml.js

```diff
@@ -278,15 +305,61 @@ var hyperHTML = (function () {'use strict';
   // Helpers
   // -------------------------
 
-  // it does exactly what it says
-  function appendNodes(node, childNodes) {
-    for (var
-      i = 0,
-      length = childNodes.length;
-      i < length; i++
-    ) {
-      node.appendChild(childNodes[i]);
+  // given a generic text or comment node
+  // remove the surrounding emptiness
+  function cleanAround(textNode, sibling) {
+    var adj = textNode[sibling];
+    if (adj && adj.nodeType === 3 && trim.call(adj.textContent).length < 1) {
+      textNode.parentNode.removeChild(adj);
+      cleanAround(textNode, sibling);
+    }
+  }
+
+  // find a live node through a virtual one
+  function find(live, virtual) {
+    var i, length, name, parentNode, map = [];
```

Comment:
nit: any reason you switched to c style variable declarations here? (i.e., why not declare i, length, in the for loop?